### DISCUSSION TITLE
Questions about the evaluation method

### DISCUSSION THREAD

**Discussion by zbzhu** [PaddlePaddle/PaddleSpatial#23] [BaiduKDDCup2022] [answered]
After reading the report and baseline code, I have some doubts about the evaluation.

1. The target variable is Patv in the provided dataset, and it's in Kilo Watt. However, in the baseline code, the predictions and true values are converted to Mega Watt before computing the evaluation score. I'm wonder why the baseline code converts Kilo Watt to Mega Watt. And **_which one should be used for the submitted results, Kilo Watt or Mega Watt?_**

2. The final score is computed on the whole test dataset by the rolling window strategy. However, **_the computation process described in the report is different from the baseline code. Which one should be used?_**
    a. According to the report, at each time point, the evaluation score of the T+288 predictions is the average of RMSE and MAE. Then, the final score is the sum of the scores of all time point in the test dataset. That is, if the whole prediction result is **X**, which has **n** rows and **288** columns, we should calculate the average of RMSE and MAE for each row firstly, and then sum all **n** scores.
    b. However, in the baseline code, the RMSE and MAE are just calculated on the whole **X**, which doesn't compute on the **n** rows separately.

**Comment by ramber1836** [comment, marked as answer]
1. Mega Watt is adopted for submitted results. 
2. Apologies for the confusion. Please take the evaluation method in the baseline code as the golden measurement. The report will be updated shortly. BTW, since the missing/unknown value issues, the evaluation method in the baseline code has been already updated accordingly.

**Reply by zbzhu** [reply]
Thanks for your reply. 
1. According to the new evaluation method in the updated code, the target variables are the wind power of 134 turbines, not the wind power of the wind farm. Is that right?
2. How to evaluate in the testing stage? Will there be a final testing dataset and we submit the forecasting results, or we just submit the prediction model?

Best regards.

**Reply by ramber1836** [reply]
1. Yes. The submitted prediction results of the wind power of 134 turbines are recommended. If someone forecasts the power of the whole wind farm, the average power of each turbine is utilized to evaluate the performance.
2. The prediction model and the related code are required.

**Reply by havetry** [reply]
> 1. Yes. The submitted prediction results of the wind power of 134 turbines are recommended. If someone forecasts the power of the whole wind farm, the average power of each turbine is utilized to evaluate the performance.
> 2. The prediction model and the related code are required.


I also want to ask a question. The basic code mentions that power conversion is required, but that line of code is commented, so whether to convert it or not?

**Reply by ramber1836** [reply]
Sorry for such confusing code, the conversion is indeed adopted, pls refer to line 177 in the metrics.py

**Reply by havetry** [reply]
> Sorry for such confusing code, the conversion is indeed adopted, pls refer to line 177 in the metrics.py

Thanks  for your reply，But I still don't understand the treatment of missing values and 0 values in the evaluation.py, I checked line 87 of the evaluation.py, he directly calls the metrics.regressor_detailed_scores function, but this function only provides the processing of unknown values, and does not consider the processing of missing values and 0 values.Is this treatment not perfect?

**Comment by ramber1836** [comment]
No, such treatment is not perfect. 
The current baseline code does not handle zero values, which will be updated accordingly in the next version dedicated for final evaluation.

**Comment by bingblackbean** [comment]
> 2\. .

The report will be updated "shortly". Do  you mean to update it before submission release or after it?
